extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php

```diff
@@ -22,24 +35,39 @@ public static function newFromPageID( $pageid ) {
 		return new PortableInfoboxDataService( Title::newFromID( $pageid ) );
 	}
 
+	// set internal helpers methods
+	public function setTemplatesHelper( $helper ) {
+		$this->templateHelper = $helper;
+
+		return $this;
+	}
+
+	public function setPagePropsProxy( $proxy ) {
+		$this->propsProxy = $proxy;
+
+		return $this;
+	}
+
 	/**
-	 * Returns infobox data
+	 * Returns infobox data, terminal method
 	 *
 	 * @return array in format [ 'data' => [], 'sources' => [] ] or [] will be returned
 	 */
 	public function getData() {
-		if ( $this->title && $this->title->exists() ) {
-			$data = $this->getParsedInfoboxes( $this->title );
-
-			//return empty [] to prevent false on non existing infobox data
-			return $data ? $data : [ ];
+		if ( $this->title && $this->title->exists() && $this->title->inNamespace( NS_TEMPLATE ) ) {
+			$hidden = $this->templateHelper->parseInfoboxes( $this->title );
```

Comment:
Ok, got it, hidden inside includeonly tags... I wonder if it can be more explicit so on encountering this line it will be immediately clear